### DISCUSSION TITLE
schema-location-pattern does not include schemas from another .jar library files

### DISCUSSION THREAD

**Discussion by khteh** [graphql-java-kickstart/graphql-spring-boot#183] [Q&A]
I have put all the .graphqls in a common library built as .jar file and references in other projects deployed as .war files. When the application starts I see the following exception:
```
Caused by: java.lang.IllegalStateException: No graphql schema files found on classpath with location pattern '**/*.graphqls'.  Please add a graphql schema to the classpath or add a SchemaParser bean to your application context.
```
Here is my application.yml:
```
graphql:
  servlet:
    enabled: true
    corsEnabled: true
    exception-handlers-enabled: true
    mapping: /graphql
  tools:
    schema-location-pattern: "classpath*:**/*.graphqls"
    #schema-location-pattern: "**/*.graphqls"
        # Enable or disable the introspection query. Disabling it puts your server in contravention of the GraphQL
        # specification and expectations of most clients, so use this option with caution
    introspection-enabled: true
```

**Comment by oliemansm** [comment]
In your config you've set the value for `schema-location-pattern` to include "classpath*", while this is already done in the code. The default schema location pattern should already handle this. I see you also include a lot of default settings in your config that you don't have to specify explicitly. Could you try if the following config works for you?
```
graphql:
  servlet:
    exception-handlers-enabled: true
```
Think all other settings are the same as the defaults.

**Comment by oliemansm** [comment]
I'm assuming my last comment helped you in fixing the problem. If not and you have any follow-up questions feel free to re-open the issue.

**Comment by brayan-bustillos-seagull-com** [comment]
Hello Michiel, is there a possibility to contact to you to talk about this problem, I;m working with spring-boot and doing a simple "hello world" but I still getting this error: "No graphql schema files found on classpath with location pattern '**/*.graphqls'"
Well only if you have the possibility to do it. Thank you.

**Comment by DineshDhyani** [comment]
Is this issue resolved? I am still facing this issue with Spring Boot GraphQL

**Comment by NuclleaR** [comment]
I added 2 dependencies to my Gradle file
```
implementation 'com.graphql-java-kickstart:graphql-spring-boot-starter:6.0.1
implementation 'com.graphql-java-kickstart:graphiql-spring-boot-starter:6.0.1'
```
And when trying to start app, app crashes with exception:
```
Caused by: java.lang.IllegalStateException: No graphql schema files found on classpath with location pattern '**/*.graphqls'.  Please add a graphql schema to the classpath or add a SchemaParser bean to your application context.
```
What I missed?

**Comment by saddis-r7** [comment]
Still not sure of the solution here, how can schema-location-pattern be customised to allow for the schema to come from a jar/dsl

**Comment by oliemansm** [comment]
First step will be adding a unit test to the library to reproduce this problem. Any contributions for this are very welcome.

**Comment by pfnectarie** [comment]
**The ONLY solution is to deploy your application as exploded WAR in Weblogic. I assume you already prepared your spring boot application for deployment on Weblogic, there are several tricks on this topic but they are available on the Internet (exclude embedded tomcat and specify WAR packaging in POM.XML, create webapp/WEB-INF folder with weblogic.xml and servlet xml).

Now create a folder on the server, /home/oracle/wars/myapp for example, and just copy the unpacked contents of your WAR to that folder.

Then go to Deployments - Install, navigate to /home/oracle/wars/ and choose myapp folder from the radio button list, and simply follow the default steps to deploy.

Pay attention, you need a plain web.xml file along with weblogic.xml, otherwise Weblogic will not see it as 'Web application'**

Deploying as exploded WAR is a workaround to avoid the creation of _wl_cls_gen.jar which is at the source of this problem.


Adding in application.properties 
graphql.tools.schema-location-pattern=**/*.graphqls
OR 
graphql.servlet.exception-handlers-enabled=true

did not help.

**Comment by rdosifyndna** [comment]
Default classpath location is set to "classpath:graphql/**/"

PathMatchingResourcePatternResolver does not pick up from jars root on classpath due to above assuming the rootdir to be "classpath:graphql/" instead of "classpath:".

Following configuration worked for me 
`spring:
  graphql:
    schema:
      locations: classpath*:graphql/**/
`

**Comment by pfnectarie** [comment]
You’re very welcome 😊

**Reply by bossm0n5t3r** [reply]
I just found more related issue (https://github.com/spring-projects/spring-boot/issues/31772), and I accidentally deleted the comment. I am so sorry. I won't do that from now on.
